overlord/ifacestate/ifacemgr.go

```diff
@@ -73,54 +73,116 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, extra []interfa
+	initialContext := map[string]interface{}{"connect-task": connectInterface.ID()}
+
 	plugHookSetup := &hookstate.HookSetup{
 		Snap:     plugSnap,
 		Hook:     "prepare-plug-" + plugName,
 		Optional: true,
 	}
-	summary := fmt.Sprintf(i18n.G("Prepare connection of plug %s:%s"), plugSnap, plugName)
-	preparePlugConnection := hookstate.HookTask(s, summary, plugHookSetup, nil)
+	summary = fmt.Sprintf(i18n.G("Prepare connection of plug %s:%s, hook %s"), plugSnap, plugName, plugHookSetup.Hook)
+	preparePlugConnection := hookstate.HookTask(s, summary, plugHookSetup, initialContext)
 
 	slotHookSetup := &hookstate.HookSetup{
 		Snap:     slotSnap,
 		Hook:     "prepare-slot-" + slotName,
 		Optional: true,
 	}
-	summary = fmt.Sprintf(i18n.G("Prepare connection of slot %s:%s"), slotSnap, slotName)
-	prepareSlotConnection := hookstate.HookTask(s, summary, slotHookSetup, nil)
+	summary = fmt.Sprintf(i18n.G("Prepare connection of slot %s:%s, hook %s"), slotSnap, slotName, slotHookSetup.Hook)
+
+	prepareSlotConnection := hookstate.HookTask(s, summary, slotHookSetup, initialContext)
 	prepareSlotConnection.WaitFor(preparePlugConnection)
 
-	summary = fmt.Sprintf(i18n.G("Connect %s:%s to %s:%s"),
-		plugSnap, plugName, slotSnap, slotName)
-	connectInterface := s.NewTask("connect", summary)
 	connectInterface.Set("slot", interfaces.SlotRef{Snap: slotSnap, Name: slotName})
 	connectInterface.Set("plug", interfaces.PlugRef{Snap: plugSnap, Name: plugName})
+	attrs, _ := initialConnectAttributes(s, plugSnap, plugName, slotSnap, slotName)
+	if attrs != nil {
+		connectInterface.Set("attributes", attrs)
+	}
 	connectInterface.WaitFor(prepareSlotConnection)
 
 	confirmSlotHookSetup := &hookstate.HookSetup{
 		Snap:     slotSnap,
 		Hook:     "connect-slot-" + slotName,
 		Optional: true,
 	}
-	summary = fmt.Sprintf(i18n.G("Confirm connection of slot %s:%s"), slotSnap, slotName)
-	confirmSlotConnection := hookstate.HookTask(s, summary, confirmSlotHookSetup, nil)
+	summary = fmt.Sprintf(i18n.G("Confirm connection of slot %s:%s, hook %s"), slotSnap, slotName, confirmSlotHookSetup.Hook)
+	confirmSlotConnection := hookstate.HookTask(s, summary, confirmSlotHookSetup, initialContext)
 	confirmSlotConnection.WaitFor(connectInterface)
 
 	confirmPlugHookSetup := &hookstate.HookSetup{
 		Snap:     plugSnap,
 		Hook:     "connect-plug-" + plugName,
 		Optional: true,
 	}
-	summary = fmt.Sprintf(i18n.G("Confirm connection of plug %s:%s"), plugSnap, plugName)
-	confirmPlugConnection := hookstate.HookTask(s, summary, confirmPlugHookSetup, nil)
+	summary = fmt.Sprintf(i18n.G("Confirm connection of plug %s:%s, hook %s"), plugSnap, plugName, confirmPlugHookSetup.Hook)
+	confirmPlugConnection := hookstate.HookTask(s, summary, confirmPlugHookSetup, initialContext)
 	confirmPlugConnection.WaitFor(confirmSlotConnection)
 
+	connectInterface.Set("confirm-plug-task", confirmPlugConnection.ID())
+	connectInterface.Set("confirm-slot-task", confirmSlotConnection.ID())
```

Comment:
s/confirm/connect/, but we actually need to review the hook up between tasks so it's a bit simpler. We probably don't need all this cross-wiring between them. We can probably keep all the attribute data on the connect task itself.